tests/pocket_test.cpp

```diff
@@ -0,0 +1,832 @@
+#include "ammo.h"
+#include "item.h"
+#include "item_pocket.h"
+
+#include "catch/catch.hpp"
+
+// Pocket Tests
+// ------------
+//
+// These tests are focused on the `item_pocket` and `pocket_data` classes, and how they implement
+// the "pocket_data" section of JSON item definitions (as documented in doc/JSON_FLAGS.md).
+//
+// To run all tests in this file:
+//
+//      tests/cata_test [pocket]
+//
+// Other tags used include: [holster], [magazine], [airtight], [watertight], [rigid], and more.
+//
+// Most of the items used for testing here are defined in data/mods/TEST_DATA/items.json, to have
+// a predictable set of data unaffected by in-game balance changes.
+
+
+// TODO: Add tests focusing on these JSON pocket_data fields:
+// -  pocket_type
+// -  spoil_multiplier
+// -  weight_multiplier
+// -  magazine_well
+// -  fire_protection
+// -  open_container
+// -  sealed_data
+// -  moves
+
+// FIXME: The failure messaging from these helper functions is pretty terrible, due to the multiple
+// CHECKs in each function. Capturing it.tname() helps only slightly. The `rate_can.value()` is an
+// integer, so it is difficult to see when a failure reason is returned instead of success.
+
+// Expect pocket.can_contain( it ) to be successful (contain_code::SUCCESS)
+static void expect_can_contain( item_pocket &pocket, item &it )
```

Comment:
```suggestion
static void expect_can_contain( const item_pocket &pocket, item &it )
```
can_contain() is a const function, might as well make this const too